src/puppetlabs/trapperkeeper/authorization/acl.clj

```diff
@@ -1,18 +1,19 @@
 (ns puppetlabs.trapperkeeper.authorization.acl
-  (:require [schema.core :as schema]
-            [clojure.string :as str]))
+  (:require [schema.core :as schema]))
 
 ;; Schemas
 
-(def auth-type (schema/enum :allow :deny))
+(def AuthType (schema/enum :allow :deny))
+(def EntryPattern (schema/conditional
+                   string? String
+                   sequential? [String]
+                   :else schema/Regex))
```

Comment:
Is `sequential? [schema/Regex]` not possible?  From an end user config perspective, I think you can have a list of regexes in an allow or deny ACL but I can't remember now how that maps onto the existing code.